### PR TITLE
Switching from Puppeteer to Playwright

### PR DESCRIPTION
**Pull request by aidanm3341**
Switch to using Playwright from Puppeteer for compatibility reasons. 
Also added logging to the visualizer.

### CODE REVIEW

**Review by lbulanti-ms — APPROVED** [review]

**Review by willosborne — APPROVED** [review]

**Review by jpgough-ms — APPROVED** [review]

**Comment by jpgough-ms** [comment]
@aidanm3341 looks like the branch has conflicts (FYI if waiting for another review)

**Comment by aidanm3341** [comment]
I have pushed a commit which fixed the merge conflicts, however GitHub is saying that "the recent push is still being processed" for some reason

**Comment by aidanm3341** [comment]
This PR appears to be in a broken state. I am going to close and create a new one.

**Pull request closed without merging**
